mantidproject/mantid
# IndirectFitDataPresenter is no longer a QObject

### Description of work
This PR refactors the IndirectFitDataPresenter, ConvFitDataPresenter and FqFitDataPresenter  so that it no longer has a QObject dependency. This makes it easier to test the presenter, and also makes it possible to replace the presenter in the future if necessary.  This also means it is easier to write a new model that conforms to a particular interface and replace the old one in the future, easily, if necessary.

Part of #36325

### To test:
1. Load the following data into the Data Analysis IqtFit tab using `Add workspace`

Input file: [irs26176_graphite002_iqt.zip](https://github.com/mantidproject/mantid/files/2361062/irs26176_graphite002_iqt.zip)
2. `Stretched Exponential`
3. `Flat background`
4. Drag the EndX on the graph to the left a bit to miss the noise (`0.15` I think)
5. Click Run and wait
6. Remove a spectrum in the top table
7. Change the EndX in the top table, and make sure the plot updates for the given spectrum
8. Attempt to unify the range across multiple spectra.

Also do a similar thing for the following tabs:

**ConvFit Tab**
Input files: [irs26176_graphite002_red.zip](https://github.com/mantidproject/mantid/files/2361070/irs26176_graphite002_red.zip)
[irs26173_graphite002_res.zip](https://github.com/mantidproject/mantid/files/2361071/irs26173_graphite002_res.zip)
Fit Type: `Two Lorenztian`
`Flat Background`

**F(Q) Tab**
Input file: [IN16B_125878_QLd_Result.zip](https://github.com/mantidproject/mantid/files/2361073/IN16B_125878_QLd_Result.zip)
Fit Type: `Chudley Elliott`

### Reviewer

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

#### Code Review

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Do the release notes conform to the [release notes guide](https://developer.mantidproject.org/Standards/ReleaseNotesGuide.html)?

#### Functional Tests

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### Gatekeeper

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.
